evmcore/tx_pool.go

```diff
@@ -677,7 +699,23 @@ func (pool *TxPool) validateTx(tx *types.Transaction, local bool) error {
 
 		gasSubsidies: pool.chain.GetCurrentRules().Upgrades.GasSubsidies,
 	}
-	err := validateTx(tx, opts, netRules, pool.chain, pool.currentState, pool.signer)
+
+	subsidiesChecker := pool.subsidiesCheckerFactory(
```

Comment:
this values do not change per transaction. Perhaps it is enough to refresh it per reset ?